src/alert_manager.cpp

```diff
@@ -71,16 +70,14 @@ namespace libtorrent
 		return NULL;
 	}
 
-	void alert_manager::maybe_notify(alert* a, mutex::scoped_lock& lock)
+	void alert_manager::maybe_notify(alert* const a)
 	{
 		if (a->type() == save_resume_data_failed_alert::alert_type
 			|| a->type() == save_resume_data_alert::alert_type)
 			++m_num_queued_resume;
 
 		if (m_alerts[m_generation].size() == 1)
 		{
-			lock.unlock();
```

Comment:
the reason to unlock the mutex before calling the user function is to avoid a deadlock in case the user function makes a call into libtorrent that in turn tries to post an alert (or pop alerts).

I would expect a call to `pop_alerts()` would deadlock with this patch for instance, if not, do you understand why?